app/models/location.rb

```diff
@@ -1,8 +1,8 @@
 class Location < ApplicationRecord
   has_many :call_to_actions
 
-  validates_presence_of :address, :city, :state, :zipcode
-  validate :unique_location
+  validates_presence_of :address
+  validate :unique_location, :sufficient_location_data_present
```

Comment:
Also you can pass `scope:` as an argument to `validates :address` uniqueness, like so:
```ruby
vaidates :address, uniqueness: { scope: [:city, :state, :zipcode] }
```

Which _should_ remove the need for the `unique_location` custom validation.

They're functionally equivalent, and I'm fine with either, so I leave it up to you @bnd5k to determine which you think is more straightforward. Hope this helps! :)